CHANGELOG.md

```diff
@@ -4,6 +4,7 @@
 
 * Update [versioning policy for plugins](https://docs.reqnroll.net/latest/installation/compatibility.html#versioning-policy) and set plugin dependencies accordingly (#160)
 * Generate symbol packages, use deterministic build and update package metadata (#161)
+* Separate `TestThreadContext` from `ITestRunner` and handle `ITestRunner.TestWorkerId` test framework independently
```

Comment:
I would probably say something that is more approachable by the end users. While testing the PR, I also noticed that currently for MsTest the previous implementation created an unnecessary test thread for each feature, even without parallel execution. 
So the changelog entry could be something like: 
`Optimize creation of test-thread context using test framework independent resource pooling (#144)`

Reply:
I changed the changelog entry 🙂